sycl/test/basic_tests/accessor/iterator-member-types.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -c %s
+// RUN: %clangxx -fsycl -fsyntax-only %s
```

Comment:
Note for reviewers: `-c` still creates an output file, which is a fat object file in case of `-fsycl` compilation and that temporary file may still cause races. Therefore, I switched `-c` tests to `-fsyntax-only`: those tests do not check LLVM IR anyway and `-fsyntax-only` doesn't create output files, which should make them a tiny bit faster (less I/O)